eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -214,4 +181,66 @@ stages:
                           PRBranchName: increment-package-version-${{ parameters.ServiceDirectory }}-$(Build.BuildId)
                           CommitMsg: "Increment package version after release of ${{ artifact.name }}"
                           PRTitle: "Increment version for ${{ parameters.ServiceDirectory }} releases"
-                          CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
\ No newline at end of file
+                          CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
+
+  - ${{if eq(variables['System.TeamProject'], 'internal') }}:
+    - stage: Integration
+      dependsOn: Build
+      condition: and(succeeded(), or(eq(variables['PublishNightlyVcpkg'], 'true'), eq(variables['Build.Reason'],'Schedule')))
+      jobs:
+        - job: PublishNightlyVcpkg
+          displayName: Publish to vcpkg nightly branch
+
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: MMS2019
+
+          steps:
+            - checkout: self
+            - download: current
+              artifact: packages
+
+            - template: /eng/pipelines/templates/steps/generate-nightly-branch-name.yml
+
+            - template: /eng/pipelines/templates/steps/vcpkg-clone.yml
+              parameters:
+                RepoOwner: azure-sdk
+                PRBranchName: $(NightlyBranchName)
+
+            # Placing artifacts in the same branch in sequence reduces the
+            # potential for race conditions. If this job does fail to push
+            # because another job has already pushed to the same branch it
+            # can be re-run.
+            - ${{ each artifact in parameters.Artifacts }}:
+              # Only run vcpkg publish if the artifact has a "VcpkgPortName"
+              # property. Absence of VcpkgPortName implies that the artifact 
+              # does not ship to vcpkg.
+              - ${{ if ne(artifact.VcpkgPortName, '') }}:
+                - template: /eng/pipelines/templates/steps/vcpkg-publish.yml
+                  parameters:
+                    ArtifactName: ${{ artifact.Name }}
+                    VcpkgPortName: ${{ artifact.VcpkgPortName }}
+                    DisplayNameExtension: ${{ artifact.Name }}
+                    TestRelease: true
+                    TestReleaseRef: $(Build.SourceVersion)
+
+                # Reset commit after vcpkg-publish.yml so git-push-changes.yml
+                # has changes to commit and push, and all changes go into one
+                # commit
+                - pwsh: |
+                    Write-Host "git reset HEAD^"
```

Comment:
Lets move this into Update-VcpkgPort.ps1. Lets keep the commit and reset in the same place.

Reply:
Looks like you are still doing a git reset here. After having a closer look at Update-VcpkgPort.ps1 it looks like you already to a reset for the vpkg`x-add-version` stuff, but you do another commit when not in test mode that will set the commit message based on the change log entry. If that is a scenario we want then I don't think you want to reset that commit, but you will need to set HasChanges=true.